ui/app/services/api.ts

```diff
@@ -154,4 +164,12 @@ export default class ApiService extends Service {
 
     return { headers };
   }
+
+  // convenience method for updating the query params object on the request context
+  // eg. this.api.sys.uiConfigListCustomMessages(true, ({ context: { query } }) => { query.authenticated = true });
+  // -> this.api.sys.uiConfigListCustomMessages(true, (context) => this.api.addQueryParams(context, { authenticated: true }));
+  addQueryParams(requestContext: { init: HTTPRequestInit; context: RequestOpts }, params: HTTPQuery = {}) {
+    const { context } = requestContext;
+    context.query = { ...context.query, ...params };
+  }
```

Comment:
Not sure that we will need this since changes to the backend to add missing params to the spec are relatively low effort but I'm going to leave it in for now just in case.